Feat/dark theme

emp74ark opened the pull request:
[Pull Request Guidelines](https://github.com/rolling-scopes/rsschool-app/blob/master/CONTRIBUTING.md#pull-requests)

**Issue**:
[Application: Dark Mode](https://github.com/rolling-scopes/rsschool-app/issues/1409)

**Description**:
- Add a theme provider that handles theme changes based on the global system theme.
- Replace inline color definitions with theme variables.
- Adapt SVG images to application theme changes.
- Update `antd` package version up to `5.26.4`
**Self-Check**:
- [ + ] Changes tested locally

[Screencast_20250709_135320.webm](https://github.com/user-attachments/assets/06d02f32-c37a-4a32-bb83-85b71798d041)

aliakseitokarev reviewed (commented):
@emp74ark, few problems left or more =)
We have to choose which icon to use depending on the theme, as it is done on the rolling-scopes [site](https://github.com/rolling-scopes/site)*: iOS, Node.js 

![Screenshot 2025-07-10 001849](https://github.com/user-attachments/assets/34032113-bb87-4922-9133-e75ab1690e3b)

Card color the same way.
![image](https://github.com/user-attachments/assets/ded2c926-549f-476a-8104-95045d144c46)

GitHub icon
![image](https://github.com/user-attachments/assets/90541ec5-2a85-4658-9aa4-f9dd79a61dd0)

Criteria for Cross-Check text color  and flash notification styles
![image](https://github.com/user-attachments/assets/d33aa527-c8ef-40e4-81fe-617c66eda63d)

emp74ark commented:
@aliakseitokarev
1. FIXED: Icon colors. The issue was resolved in a slightly unconventional way because the icons are stored in the public folder and used as external resources.
2. FIXED: Notification and message colors. These methods from Ant Design were used outside the theme context. While the problem itself is simple, fixing it caused changes in many files. I am not sure if I can manually verify all of them, but all tests have passed.
3. FIXED: CV view background.
4. FIXED: Course Events description background.

I couldn't reproduce the white color issue of the search input in `/admin/users`.
If possible, could you share an example of the JSON for Cross-Check criteria?

emp74ark commented on client/public/static/svg/disciplines/ios.svg:
`width="100%" height="100%" viewBox="0 0 14 15"`
Icon size depends on the bock size where they are used. There are no fixed dimensions.

pestler replied:
ok

aliakseitokarev commented:
> I couldn't reproduce the white color issue of the search input in `/admin/users`.


https://github.com/user-attachments/assets/13f6fd6d-b784-41b1-bed3-f7b9e943dde8

aliakseitokarev commented:
> > > Didn't understand message: `Card color the same way.` What is wrong with this card?
> > 
> > 
> > I mean, we need to change the background color of the card. It should be a more appropriate color for the dark theme; the current color is too bright
> 
> Well, I'm not good at choosing perfect colors. Any suggestions?


https://github.com/user-attachments/assets/c4843260-e83b-4ea0-8c7b-55b3c8a2b8cd

aliakseitokarev commented:
> 1. FIXED: Icon colors. The issue was resolved in a slightly unconventional way because the icons are stored in the public folder and used as external resources.

There is still a small bug on the main page in the selector

<img width="1153" height="568" alt="image" src="https://github.com/user-attachments/assets/16832126-a36e-43a5-87ed-9527ad526798" />

aliakseitokarev commented:
> Please share the url of this page

Home page (main page)

emp74ark commented:
> > > > Didn't understand message: `Card color the same way.` What is wrong with this card?
> > > 
> > > 
> > > I mean, we need to change the background color of the card. It should be a more appropriate color for the dark theme; the current color is too bright
> > 
> > 
> > Well, I'm not good at choosing perfect colors. Any suggestions?
> 
> Me too! Let's use the automatic ')
> 
>  Recording.2025-07-11.161223.mp4

Please, check now.

AlreadyBored commented:
@emp74ark thank you for your work! From what I see it works correctly, however, I have some suggestions:
1. It makes sense to store selected theme somewhere (at least in localstorage). Because if you refresh page, theme is reset to default, this is not good from UX perspective.
2. Please avoid changes that are not related with main scope of PR, like changing of formatting of object props (like [here](https://github.com/rolling-scopes/rsschool-app/pull/2752/files#diff-cce91948574755f5b518eb0d29f02961141126cbb8e724f767c208ae89060284R230) or [here](https://github.com/rolling-scopes/rsschool-app/pull/2752/files#diff-cce91948574755f5b518eb0d29f02961141126cbb8e724f767c208ae89060284R204), etc.). These changes make PR bigger and harder to review.

If you want to change formatting – please consider changing linter/formatter config and making these changes part of it, so it could be automatically applied to all our codebase + implement it as separated PR.

emp74ark commented:
> @emp74ark thank you for your work! From what I see it works correctly, however, I have some suggestions:
> 
> 2. Please avoid changes that are not related with main scope of PR, like changing of formatting of object props (like [here](https://github.com/rolling-scopes/rsschool-app/pull/2752/files#diff-cce91948574755f5b518eb0d29f02961141126cbb8e724f767c208ae89060284R230) or [here](https://github.com/rolling-scopes/rsschool-app/pull/2752/files#diff-cce91948574755f5b518eb0d29f02961141126cbb8e724f767c208ae89060284R204), etc.). These changes make PR bigger and harder to review.
> 
> If you want to change formatting – please consider changing linter/formatter config and making these changes part of it, so it could be automatically applied to all our codebase + implement it as separated PR.

Frankly saying I didn't change formatting manually - I use npm script from the project root to avoid `pr_lint` job warnings.

aliakseitokarev commented on client/src/modules/Schedule/components/TableView/TableView.test.tsx:
```diff
@@ -172,7 +168,11 @@ describe('TableView', () => {
       // Mock useLocalStorage for combinedFilter
       .mockReturnValueOnce([
-        { types: [TagsEnum.Coding, TagsEnum.Test, TagsEnum.Interview], statuses: [], tags: [] },
+        {
+          types: [TagsEnum.Coding, TagsEnum.Test, TagsEnum.Interview],
+          statuses: [],
+          tags: [],
+        },
```
```suggestion
        { types: [TagsEnum.Coding, TagsEnum.Test, TagsEnum.Interview], statuses: [], tags: [] },
```

aliakseitokarev commented on client/src/modules/Schedule/components/TableView/TableView.test.tsx:
```diff
@@ -209,7 +209,11 @@ describe('TableView', () => {
           types,
           statuses: [],
-          tags: types.map(t => ({ label: `${ColumnName.Type}: ${t}`, value: t, tagType: ColumnName.Type })),
+          tags: types.map(t => ({
+            label: `${ColumnName.Type}: ${t}`,
+            value: t,
+            tagType: ColumnName.Type,
+          })),
```
```suggestion
          tags: types.map(t => ({ label: `${ColumnName.Type}: ${t}`, value: t, tagType: ColumnName.Type })),
```

aliakseitokarev commented on client/src/modules/TeamDistribution/components/WelcomeCard/WelcomeCard.tsx:
```diff
@@ -26,7 +31,15 @@ export default function WelcomeCard({ isManager, handleCreateTeamDistribution }:
         </Col>
         {!mobileView && (
-          <Col md={12} style={{ position: 'relative', display: 'flex', justifyContent: 'end', paddingRight: '24px' }}>
+          <Col
+            md={12}
+            style={{
+              position: 'relative',
+              display: 'flex',
+              justifyContent: 'end',
+              paddingRight: '24px',
+            }}
+          >
```
```suggestion
          <Col md={12} style={{ position: 'relative', display: 'flex', justifyContent: 'end', paddingRight: '24px' }}>
```

aliakseitokarev commented on client/src/components/Student/AssignStudentModal.tsx:
```diff
@@ -32,7 +34,8 @@ export function AssignStudentModal(props: Props) {
       title={
         <>
-          <Text>Assign Student to</Text> <Text underline>{props.mentorGithuId}</Text>
+          <Text>Assign Student to</Text>
+          <Text underline>{props.mentorGithuId}</Text>
```
```suggestion
          <Text>Assign Student to</Text> <Text underline>{props.mentorGithuId}</Text>
```

aliakseitokarev commented on client/src/modules/Mentor/pages/Students/index.tsx:
```diff
@@ -1,9 +1,9 @@
 import {
+  InteractionTwoTone,
+  LockFilled,
```
No need to change this file
